Content.Server/GameObjects/Components/Gravity/GravityGeneratorComponent.cs

```diff
@@ -100,27 +101,26 @@ bool IAttackHand.AttackHand(AttackHandEventArgs eventArgs)
 
         public bool AttackBy(AttackByEventArgs eventArgs)
         {
-            if (!eventArgs.AttackWith.TryGetComponent<WelderComponent>(out var welder)) return false;
-            if (welder.TryUse(5.0f))
-            {
-                // Repair generator
-                var damagable = Owner.GetComponent<DamageableComponent>();
-                var breakable = Owner.GetComponent<BreakableComponent>();
-                damagable.HealAllDamage();
-                breakable.broken = false;
-                _intact = true;
+            if (!eventArgs.AttackWith.TryGetComponent(out WelderComponent tool))
+                return false;
 
-                var entitySystemManager = IoCManager.Resolve<IEntitySystemManager>();
-                var notifyManager = IoCManager.Resolve<IServerNotifyManager>();
+            if (!tool.UseTool(eventArgs.User, Owner, ToolQuality.Welding, 5f))
```

Comment:
> it would be neat if this showed a notification like "your welder is empty/off" if applicable.

I just added this in WelderComponent, making it show popups explaining why you couldn't use it. 

> `UseTool` doesn't distinguish between "this tool doesn't have this quality" and "this tool has this quality but is empty/off". Should it?

I personally think it shouldn't. The tool should handle this, showing a popup/notification to the user like you suggested and I added.

> I would expect in many cases for `AttackBy` to be handled when using an empty welder but then there's edge cases like closets where you probably don't (want to put welder into closet instead of deconstructing closet).

Hmmm, I'm not too sure what would be the best way to handle that.